testhelpers/fake_firehose.go

```diff
@@ -100,4 +100,9 @@ func (f *FakeFirehose) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 			panic(err)
 		}
 	}
+
+	// wait a bit before closing the connection with the nozzle
+	// and forcing it to call PostMetrics
+	// this gives the nozzle time to process the envelopes we just sent
+	time.Sleep(500 * time.Millisecond)
```

Comment:
🍰 this is a good fix--no need to overengineer the test

Reply:
Fair enough!